### ISSUE TITLE
Incorrect detection/flagging of rule `Moq1203` after upgrading to `v0.4.0`

### ISSUE THREAD

**Issue by AgustinSabalza** [rjmurillo/moq.analyzers#849]
### Description
Existing Setups break after upgrading from `0.3.1` to `0.4.0` due to incorrect flagging of **Moq1203**.

### Steps To Reproduce
``` csharp
public class Moq1203Tests
{
    public interface IFoo
    {
        Task<int> BarAsync();
    }

    [Fact]
    public void Test_Moq1203()
    {
        var moq = new Mock<IFoo>(MockBehavior.Strict);

        // Flags Moq1203.
        moq.Setup(x => x.BarAsync())
            .ReturnsAsync(1);

        // Flags Moq1203.
        moq.Setup(x => x.BarAsync())
            .Callback(() => { })
            .Returns(Task.FromResult(1));

        // OK.
        moq.Setup(x => x.BarAsync())
            .Returns(Task.FromResult(1));

        // OK.
        moq.Setup(x => x.BarAsync())
            .Returns(Task.FromResult(1))
            .Callback(() => { });
    }
}
```

### Environment
- .NET 10
- Moq.Analyzers v0.4.0

**Comment by bkaidbb** [comment]
Can confirm upgrading to `0.4.1-alpha` fixes my issue - thanks!

**Comment by DamienCassou** [comment]
> @DamienCassou I can't get your case to repro with the new fixes. I've opened a PR that will merge to main with the repro cases from this thread which should provide a backstop. Can you check my attachment and your environment to make sure you're using the alpha package? I want to make sure your use case is unblocked.

I'm not sure how to reproduce that as the zip file doesn't include the nugets.

What about releasing a 0.4.1-alpha.1 version (following https://semver.org/)?

**Comment by rjmurillo** [comment]
> > [@rjmurillo](https://github.com/rjmurillo) oh sorry. I understand what you mean now. I took the zip file and replaced 0.5.0-alpha-g85a99feef4 with 0.4.1-alpha. I was able to get `dotnet build` to successfully build the 2 projects. I also went back to the original project I created to reproduce the issue and everything builds just fine now.
> > Unfortunately, I found another issue which reveals itself when using `ReturnsAsync((...) => ...)`:
> > using Moq;
> > 
> > namespace TestProject1.UnitTest2;
> > 
> > public record MyValue;
> > 
> > public interface IDatabase
> > {
> >     public Task<MyValue> SaveAsync(MyValue value);
> > }
> > 
> > public class Database : IDatabase
> > {
> >     public async Task<MyValue> SaveAsync(MyValue value)
> >     {
> >         await Task.CompletedTask;
> > 
> >         return value;
> >     }
> > }
> > 
> > public class UserOfDatabase(IDatabase database)
> > {
> >     public async Task<MyValue> SaveAsync(MyValue value)
> >     {
> >         Console.WriteLine(value);
> >         return await database.SaveAsync(value);
> >     }
> > }
> > 
> > public class UnitTest2
> > {
> >     [Test]
> >     public async Task Test1()
> >     {
> >         var databaseMock = new Mock<IDatabase>(MockBehavior.Strict);
> >         var userOfDatabase = new UserOfDatabase(databaseMock.Object);
> >         var input = new MyValue();
> > 
> >         databaseMock
> >             .Setup(mock => mock.SaveAsync(It.IsAny<MyValue>()))
> >             .ReturnsAsync((MyValue val) => val);
> > 
> >         var actualValue = await userOfDatabase.SaveAsync(input);
> > 
> >         Assert.That(actualValue, Is.EqualTo(input));
> >     }
> > }
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [repro.zip](https://github.com/user-attachments/files/25411062/repro.zip)
> 
> I'll check to make sure [#896](https://github.com/rjmurillo/moq.analyzers/issues/896) covers that

@DamienCassou this case will be addressed in #907  and part of the 0.4.1 stable release
